metadata/src/main/java/org/apache/kafka/controller/PartitionChangeBuilder.java

```diff
@@ -54,6 +54,7 @@ public static boolean changeRecordIsNoOp(PartitionChangeRecord record) {
         if (record.removingReplicas() != null) return false;
         if (record.addingReplicas() != null) return false;
         if (record.leaderRecoveryState() != LeaderRecoveryState.NO_CHANGE) return false;
+        if (record.assignment() != null) return false;
```

Comment:
The `org.apache.kafka.metadata.PartitionRegistration` class has a `merge()` method that will need some adjusting as well.

Reply:
I had addressed that and lots of other references to `Replicas` in #14516 

https://github.com/apache/kafka/pull/14516/files#diff-61a58fedae1bdadb73928dd41ccc4eeffe47d599557ad2b2973e3124d38e3d44R191-R196

Would it be preferable to join the two PRs in a single patch?

Reply:
> Would it be preferable to join the two PRs in a single patch?

They are related for sure.  I assume the changes specific to that other PR would be much smaller if we decided to abandon the `Assignment` of type `[]ReplicaAssignment` in favor of just an array of UUIDs.  Then the changes would just be specific to that which is required for JBOD as opposed to all the changes related to a wholesale replacement of `Replicas []int32` with `Assignment []ReplicaAssignment`.  My preference right now would be to abandon `[]ReplicaAssignment` and include the then-smaller changes from that other PR here.